Content.Client/GameObjects/Components/Research/LatheDatabaseComponent.cs

```diff
@@ -0,0 +1,58 @@
+using Content.Shared.GameObjects.Components.Research;
+using Content.Shared.Research;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Interfaces.GameObjects;
+using Robust.Shared.Interfaces.Network;
+using Robust.Shared.IoC;
+using Robust.Shared.Prototypes;
+
+namespace Content.Client.GameObjects.Components.Research
+{
+    public class LatheDatabaseComponent : SharedLatheDatabaseComponent
+    {
+#pragma warning disable CS0649
+        [Dependency]
+        private IPrototypeManager _prototypeManager;
+#pragma warning restore
+
+        public override void Initialize()
+        {
+            base.Initialize();
+            IoCManager.InjectDependencies(this);
+        }
+
+        public override void HandleMessage(ComponentMessage message, INetChannel netChannel = null, IComponent component = null)
```

Comment:
See this method of syncing prototypes is kind of problematic since this will not work if your client joins AFTER the lathe has been created.

You should use component states for this.

Reply:
Done. ComponentStates are really useful, huh. I wish I'd known about them earlier.